### PR TITLE
gl_rasterizer: Implement partial color clear and stencil clear.

### PR DESCRIPTION
**Pull request by bunnei**

### CODE REVIEW

File: src/video_core/renderer_opengl/gl_rasterizer.cpp [hunk 1]
```diff
@@ -349,10 +349,19 @@ std::pair<Surface, Surface> RasterizerOpenGL::ConfigureFramebuffers(bool using_c
 
 void RasterizerOpenGL::Clear() {
     const auto& regs = Core::System::GetInstance().GPU().Maxwell3D().regs;
-
+    const auto previous_state{state};
```
Comment (Subv) [inline src/video_core/renderer_opengl/gl_rasterizer.cpp]: Use SCOPE_EXIT to restore the previous state

File: src/video_core/renderer_opengl/gl_rasterizer.cpp [hunk 2]
```diff
@@ -349,10 +349,19 @@ std::pair<Surface, Surface> RasterizerOpenGL::ConfigureFramebuffers(bool using_c
...
     bool use_color_fb = false;
     bool use_depth_fb = false;
 
+    if (regs.clear_buffers.RT != 0) {
```
Comment (Subv) [inline src/video_core/renderer_opengl/gl_rasterizer.cpp]: add a log/assert for this

Comment (Subv) [reply src/video_core/renderer_opengl/gl_rasterizer.cpp]: This seems wrong, RT = 0 means the game doesn't want to clear any render targets, ie, a depth/stencil-only clear.

**Comment by bunnei** [comment]
Closing - I'm superseding this PR.

**Pull request closed without merging**